plan/logical_plan_builder.go

```diff
@@ -1130,10 +1131,24 @@ func (b *planBuilder) buildDataSource(tn *ast.TableName) LogicalPlan {
 			TblName:  tableInfo.Name,
 			DBName:   schemaName,
 			RetType:  &col.FieldType,
-			Position: i,
+			Position: i, // NOTE: Position can be discontinuous?
 			ID:       col.ID})
 	}
 	p.SetSchema(schema)
+	// for not stored generated column
+	for _, col := range schema.Columns {
+		idx := col.Position
+		colInfo := tableInfo.Columns[idx]
+		if len(colInfo.GeneratedExprString) != 0 && !colInfo.GeneratedStored {
+			column := tbl.Cols()[idx]
+			expr, _, err := b.rewrite(column.GeneratedExpr, p, nil, true)
+			if err != nil {
+				b.err = errors.Trace(err)
+				return nil
+			}
+			p.GenValues[idx] = expr
```

Comment:
After try, I think do virtual column generation in `column prune` is not a better idea, because all logical plans above `DataSource` need to add some code. That's too wordy.